### PR TITLE
feat(tree-view): simplify component api

### PR DESCRIPTION
**Pull request by cschroeter**
### Open Questions

Slack https://chakraui.slack.com/archives/C0491R77XE1/p1707299931872079

### New Component API

```jsx
    <TreeView.Root>
      <TreeView.Label>Tree</TreeView.Label>
      <TreeView.Tree>
        <TreeView.Item id="1.0">
          <TreeView.ItemText>1.0</TreeView.ItemText>
        </TreeView.Item>
        <TreeView.Branch id="2.0">
          <TreeView.BranchControl>
            <TreeView.BranchTrigger>
              <TreeView.BranchText>2.0</TreeView.BranchText>
              <TreeView.BranchIndicator>↕️</TreeView.BranchIndicator>
            </TreeView.BranchTrigger>
          </TreeView.BranchControl>
          <TreeView.BranchContent>
            <TreeView.Item id="2.1">
              <TreeView.ItemText>2.2</TreeView.ItemText>
            </TreeView.Item>
            <TreeView.Item id="2.2">
              <TreeView.ItemText>2.2</TreeView.ItemText>
            </TreeView.Item>
          </TreeView.BranchContent>
        </TreeView.Branch>
      </TreeView.Tree>
    </TreeView.Root>
  ```

### CODE REVIEW

File: packages/frameworks/react/src/tree-view/tree-view-root.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export interface TreeViewRootProps
   > {}
 
 export const TreeViewRoot = forwardRef<HTMLDivElement, TreeViewRootProps>((props, ref) => {
-  const [useTreeViewProps, { children, ...divProps }] = createSplitProps<UseTreeViewProps>()(
+  const [useTreeViewProps, { children, ...localProps }] = createSplitProps<UseTreeViewProps>()(
```
Comment (Omikorin) [inline packages/frameworks/react/src/tree-view/tree-view-root.tsx:22]: We should decide on naming this. There are many inconsistent `divProps`, `spanProps` and `localProps` etc throughout our components.

File: packages/frameworks/react/src/tree-view/tree-view-tree.tsx [hunk 2]
```diff
@@ -2,14 +2,19 @@ import { mergeProps } from '@zag-js/react'
...
 
 export interface TreeViewTreeProps extends HTMLArkProps<'ul'> {}
 
 export const TreeViewTree = forwardRef<HTMLUListElement, TreeViewTreeProps>((props, ref) => {
   const api = useTreeViewContext()
   const mergedProps = mergeProps(api.treeProps, props)
 
-  return <ark.ul {...mergedProps} ref={ref} />
+  return (
+    <TreeViewDepthProvider value={1}>
```
Comment (Omikorin) [inline packages/frameworks/react/src/tree-view/tree-view-tree.tsx:14]: I like this simple provider

File: packages/frameworks/react/src/tree-view/stories/tree-view.stories.tsx [hunk 3]
```diff
@@ -10,71 +10,27 @@ export default meta
...
-            </TreeView.BranchText>
+        <TreeView.Item id="1.0">
+          <TreeView.ItemText>1.0</TreeView.ItemText>
+        </TreeView.Item>
+        <TreeView.Branch id="2.0">
+          <TreeView.BranchControl>
+            <TreeView.BranchTrigger>
+              <TreeView.BranchText>2.0</TreeView.BranchText>
+              <TreeView.BranchIndicator>↕️</TreeView.BranchIndicator>
+            </TreeView.BranchTrigger>
```
Comment (Omikorin) [inline packages/frameworks/react/src/tree-view/stories/tree-view.stories.tsx:23]: We could show that it's possible to not use `BranchTrigger` which looks like a basic branch without button. I am not sure if this matters for WAI-ARIA in this case

image

**Review by Omikorin — APPROVED** [review]

File: packages/frameworks/react/src/tree-view/tree-view-root.tsx (hunk 1 above)
Comment (cschroeter) [reply packages/frameworks/react/src/tree-view/tree-view-root.tsx:22]: @Omikorin I would go with local props since it is interchangeable

**Pull request closed without merging**